apps/api/src/app/events/services/performance-service/index.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { IMark, PerformanceContextEnum, PerformanceService } from '../../../shared/services/performance';
+
+enum MarkFunctionNameEnum {
+  CREATE_NOTIFICATION_JOBS = 'createNotificationJobs',
+  DIGEST_FILTER_STEPS = 'digestFilterSteps',
+  ENDPOINT_TRIGGER_EVENT = 'endpoint:triggerEvent',
+  TRIGGER_EVENT = 'triggerEvent',
+}
+
+@Injectable()
+export class EventsPerformanceService {
```

Comment:
I don't think that we need to create an events-specific performance service, I feel like all this might be generic code, that can be reused in any context. This way we won't end up writing per context performance services. So I see it with the decorator as I mentioned before.

The `publishResults` function can be defined as the helper function that will be called from anywhere we need to get the stats.

Reply:
I try to think of the different modules as if they were its own microservice, so making this domain related service allows to:
- Keep isolated in the domain the specific mark generation.
- Have their own kind of calculations and what publish and what not (think in a different domain we might want to have different calculations, instead of average other algorithm, publish different results, etc)

That doesn't mean that's going to happen, to be reused elsewhere, so I can understand why you are suggesting something more generic.

Reply:
I don't know to me it would be more to maintain... but yeah maybe let's start with something simple and then adjust to the new requirements in the future...